plugins/modules/apk.py

```diff
@@ -371,6 +371,12 @@ def main():
     if p['upgrade']:
         upgrade_packages(module, p['available'])
 
+    if all(not name.strip() for name in p['name']):
```

Comment:
Why have this check here, and not further up (say, directly after `p = module.params`)?

Reply:
@gdrosos this question is still open ^

Reply:
Sorry, it seems after the rebase I accidentally deleted my response.

So I placed the check where it is now because I thought in cases where a user wants to update the repository index and supplies an empty string for the package name, e.g.:

```yaml
- name: Update repositories and update package empty
  community.general.apk:
    name: ""
    update_cache: true
```

it made more sense to first update the repository index and then handle the empty string. Placing the check earlier would skip the repository update, which I wanted to avoid. Let me know if you'd like me to adjust the order!

Reply:
If the parameters do not pass validation, I think it would make more sense not to change any state of any resource in the target.

Or, let us put it from a different perspective: with the PR, empty-ish names are not allowed and will trigger an error. When you run a playbook and one of the tasks pops an error (and a validation one at that), would you expect it to make any change at all?

I think the validation should happen at line 353 as Felix suggested.